org.palladiosimulator.analyzer.slingshot.behavior.spd/src/org/palladiosimulator/analyzer/slingshot/behavior/spd/interpreter/entity/trigger/CPUUtilizationTriggerChecker.java

```diff
@@ -0,0 +1,112 @@
+import org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.utils.TargetGroupUtils;
+import org.palladiosimulator.analyzer.slingshot.common.events.DESEvent;
+import org.palladiosimulator.analyzer.slingshot.monitor.data.events.MeasurementMade;
+import org.palladiosimulator.edp2.models.measuringpoint.MeasuringPoint;
+import org.palladiosimulator.metricspec.constants.MetricDescriptionConstants;
+import org.palladiosimulator.pcm.resourceenvironment.ProcessingResourceSpecification;
+import org.palladiosimulator.pcmmeasuringpoint.ActiveResourceMeasuringPoint;
+import org.palladiosimulator.spd.targets.TargetGroup;
+import org.palladiosimulator.spd.triggers.AGGREGATIONMETHOD;
+import org.palladiosimulator.spd.triggers.SimpleFireOnValue;
+import org.palladiosimulator.spd.triggers.expectations.ExpectedPercentage;
+import org.palladiosimulator.spd.triggers.stimuli.CPUUtilization;
+
+public class CPUUtilizationTriggerChecker extends TriggerChecker<CPUUtilization> {
+
+	/** The number of measurements to consider at most. */
+	private static final int THRESHOLD = 10;
+
+	private final AbstractWindowAggregation aggregator;
+	private final TargetGroup targetGroup;
+
+	public CPUUtilizationTriggerChecker(final SimpleFireOnValue trigger,
+										final AGGREGATIONMETHOD aggregationMethod,
+								 		final TargetGroup targetGroup) {
+		super(trigger, CPUUtilization.class, Set.of(ExpectedPercentage.class));
+
+		this.targetGroup = targetGroup;
+		this.aggregator = switch (aggregationMethod) {
+			case MIN -> new MinWindowAggregation(THRESHOLD);
+			case AVERAGE -> new MeanWindowAggregation(THRESHOLD);
+			case MAX -> new MaxWindowAggregation(THRESHOLD);
+			case MEDIAN -> new MedianWindowAggregation(THRESHOLD);
+			case SUM -> new SumWindowAggregation(THRESHOLD);
+			default -> throw new IllegalArgumentException("Unexpected value: " + aggregationMethod);
+		};
+	}
+
+	@Override
+	public FilterResult doProcess(final FilterObjectWrapper object) {
+		final DESEvent event = object.getEventToFilter();
+		if (event instanceof final MeasurementMade measurementMade) {
+			final MeasuringPoint measuringPoint = measurementMade.getEntity().getMeasuringPoint();
+			if (measuringPoint instanceof final ActiveResourceMeasuringPoint activeResourceMP) {
+				/*
+				 * Check that the measuring point points to one of the resource container's
+				 * specifications, and count them only once!
+				 */
+				aggregateMeasurement(measurementMade, activeResourceMP);
+
+
+				/*
+				 * If all the values are collected, then check whether the expected percentage
+				 * matches, otherwise disregard.
+				 */
+				return getResult(event);
+			}
+		}
+
+
+		return FilterResult.disregard("");
+	}
+
+	/**
+	 * Helper method to retrieve the filter result. If the aggregated value is in
+	 * accordance with the specified trigger, success is returned. If not all
+	 * measurements were made yet, or if the value was not in accordance, then
```

Comment:
it seems that the docu promises something that is not in the implementation or I am not able to see it: how is the part of not all measurements were made realized?

Reply:
I think this got lost over the last changes. The aggregator used to directly return the result for the comparator rather than the actual value. The idea was if the window (the queue) is not full yet, then it should wait for further measurements until it is full. So we only aggregate when the window was completely filled up.

Reply:
@kjuli provides the missing code for this